File: Data Structures & Algorithms/longest-palindromic-substring/submission-0.java
class Solution {

    public int helper(String s, int left, int right, int n){
       while(left >= 0 && right <= n - 1 && s.charAt(left) == s.charAt(right)){
            left--;
            right++;
       } 
       return right - left - 1;
    }

    public String longestPalindrome(String s) {
        int n = s.length();
        int start = 0;
        int end = 0;
        for(int i = 0; i<n;i++){
            int len1 = helper(s, i, i+1, n);
            int len2 = helper(s, i, i, n);
            int len = Math.max(len1, len2);
            if(len > end - start){
                start = i - (len-1)/2;
                end = i + len/2;
            }
        }
        return s.substring(start, end+1);
    }
}

File: Data Structures & Algorithms/valid-binary-search-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def helper(self, root: Optional[TreeNode], minValue: int, maxValue) -> bool:
        if root == None:
            return True
        if root.val <= minValue:
            return False
        if root.val >= maxValue:
            return False
        
        return self.helper(root.left, minValue, root.val) and \
                self.helper(root.right, root.val, maxValue) 

    def isValidBST(self, root: Optional[TreeNode]) -> bool:
        return self.helper(root, -math.inf, math.inf)
        
        